Kernel/Heap/kmalloc.cpp

```diff
@@ -243,6 +298,33 @@ void* kmalloc_eternal(size_t size)
     return ptr;
 }
 
+void* kmalloc_eternal(size_t size)
+{
+    kmalloc_verify_nospinlock_held();
+
+    size = round_up_to_power_of_two(size, sizeof(void*));
+
+    ScopedSpinLock lock(s_lock);
+    // Note: Only Eternal heaps should be in the static eternal heap of the kernel,
+    // therefore each pointer must of an EternalHeap and only it!
+    EternalHeap* eternal_heaps_metadata = reinterpret_cast<EternalHeap*>(kmalloc_eternal_heaps);
+    if (auto* ptr = eternal_heaps_metadata[g_dynamic_eternal_heaps_count].allocate(size); ptr != nullptr)
+        return ptr;
+
+    // Note: It's very unlikely that the MemoryManager was not initialized at this stage!
+    VERIFY(MemoryManager::is_initialized());
+
+    // Allocate a region. If the allocation fails, assert.
+    auto region = MM.allocate_kernel_region((1 * MiB), "kmalloc eternal heap", Region::Access::Read | Region::Access::Write, AllocationStrategy::AllocateNow);
+    VERIFY(region);
+    new EternalHeap(region.release_nonnull());
+    g_dynamic_eternal_heaps_count++;
+    if (auto* ptr = eternal_heaps_metadata[g_dynamic_eternal_heaps_count].allocate(size); ptr != nullptr)
+        return ptr;
+    // Note: We have a fresh new eternal heap, so there's no sane way to be out of memory with it now.
```

Comment:
What happens if you try to `kmalloc_eternal()` more than 1 MiB?

Reply:
I could a quick fix to make it to work allocations larger than 1 MiB. I think it's much better to just assert if we try to allocate more than 1 MiB at once, at least I consider such pattern to be a bug (unless there's really a good reason to allocate lots of memory after booting the kernel and setting up basic components in the init process).

Reply:
I fixed this :)